crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -332,12 +332,19 @@ where
                     .get_stage_checkpoint(*StageId::ALL.first().unwrap())?
                     .unwrap_or_default()
                     .block_number;
-                let last_stage_checkpoint = this
-                    .blockchain
-                    .get_stage_checkpoint(*StageId::ALL.last().unwrap())?
-                    .unwrap_or_default()
-                    .block_number;
-                if first_stage_checkpoint != last_stage_checkpoint {
+                let mut need_pipeline_run = false;
+                for stage_id in StageId::ALL.iter().skip(1) {
```

Comment:
why are we skipping headers?

I will forget what this does so I'd like to move this to a function and document properly

Reply:
extracted and documented. the first stage is skipped, because we've already retrieved it above and are comparing all other checkpoints against it